addons/website/static/src/js/editor/snippets.editor.js

```diff
@@ -66,5 +78,15 @@ weSnippetEditor.Class.include({
             tab: this.tabs.THEME,
         });
     },
+    /**
+     * @private
+     * @param {OdooEvent} ev
+     */
+    _onGetCustomColors: function (ev) {
+        const variables = this.style.getPropertyValue(`--custom-colors`).trim().split(' ');
+        const colors = variables.map(v => this.style.getPropertyValue(`--${v}`).trim())
+                                .filter(c => ColorpickerDialog.isCSSColor(c));
+        ev.data.onSuccess(colors);
+    },
```

Comment:
This does not seems related to the snippet editor class at all or am I mistaken? Should not this be done directly in the color palette widget ?

Reply:
These custom colors are related to the theme colors so I think this is not the wrong place. Especially since there was already a trigger up mechanism to retrieve Chart custom colors. 
Still putting it here instead of color_palette is not really that useful since color_palette will most likely be the only widget using it. So I have no problem moving it.

Reply:
How is a theme color related to the management of options of a specific snippet? I don't see the link at all personally :/ In fact, it means that if you open the color palette in the editor top bar you don't get access to text, h1, h2, etc colors... although you are in the exact same editor, editing the exact same content (a snippet) :wink: 

Seems like "the colorpalette uses the theme colors of the current page" is the feature you want (no "snippet" in that sentence). From there, there are two choices: reading the css variable from the colorpalette directly, or using the 'get_custom_colors' system... but reaching the top level element of the editor. Option 2 seems appropriate but since the top level object of the editor is subject to lots of changes in the upcoming month, let's go with the change in the colorpalette object directly.

(Note: I am still convinced you don't want access to header / footer / etc colors from anywhere though but that's another debate ^^).